### DISCUSSION TITLE
how to catch popups

### DISCUSSION THREAD

**Discussion by dongfanglong2012** [r0x0r/pywebview#826] [Q&A]
is there a way to catch popups in a new pyweview window? for example the site loaded in the pywebview has a popup when a given btn is clicked that opens a new window. Is there a way to force that new window to be a pywebview without editing the website.

**Comment by r0x0r** [comment]
No, not currently, but my feeling is that it can be implemented at least in some renderers. Pywebview catches target=blank links and opens them in an external browser though. Popups can probably be caught in a similar way.

**Reply by dongfanglong2012** [reply]
do you have any examples of that being used? I'm not sure how to go implementing something like that

**Comment by r0x0r** [comment]
The implementation varies from renderer to renderer, but if you search for webbrowser.open, you will find some examples.
E.g. edge chromium catches  NewWindowRequested  event, it might catch pop ups as well
